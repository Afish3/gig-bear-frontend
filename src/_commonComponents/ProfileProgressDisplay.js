import * as React from 'react';
import CircularProgress from '@mui/material/CircularProgress';
import Typography from '@mui/material/Typography';
import Box from '@mui/material/Box';

const ProfileProgressDisplay = ({ progress, size="12rem" }) => {
    
    return (
        <Box sx={{ position: 'relative', display: 'inline-flex' }} >
            <CircularProgress variant="determinate" 
            value={progress} 
            size={size} 
            thickness={2} 
            data-testid="progressbar" />
            <Box
            data-testid="outer-box"
            sx={{
                top: 0,
                left: 0,
                bottom: 0,
                right: 0,
                position: 'absolute',
                display: 'flex',
                alignItems: 'center',
                justifyContent: 'center',
                borderRadius: `${size}`,
                backgroundColor: 'rgba(5, 5, 5, 0.1)',
            }}
            >
          <Typography variant="caption" component="div" color="text.secondary">
            {`${Math.round(progress)}%`}
          </Typography>
        </Box>
      </Box>
      );
  }

  export default ProfileProgressDisplay;