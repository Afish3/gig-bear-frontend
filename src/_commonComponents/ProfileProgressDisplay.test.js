import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import ProfileProgressDisplay from './ProfileProgressDisplay';

describe('ProfileProgressDisplay Component', () => {
  it('renders the progress circle with the correct percentage', () => {
    render(<ProfileProgressDisplay progress={75} />);
    
    // Check if the progress text is visible
    const progressText = screen.getByText('75%');
    expect(progressText).toBeInTheDocument();

    // Check if the CircularProgress component is present
    const circularProgress = screen.getByTestId('progressbar');
    expect(circularProgress).toBeInTheDocument();

    /** Check if the CircularProgress component has the correct aria-valuenow attribute.
     * 
     * aria-valuenow attribute added by MUI for accessability and is updated to reflect the value.
     * 
     * checking the aria-valuenow attribute instead of value because value does not translate
     * to the HTML when the component renders on the DOM.
     */
    expect(circularProgress).toHaveAttribute('aria-valuenow', '75');
  });

  it('handles 0 percent progress completion', () => {
    render(<ProfileProgressDisplay progress={0} />);
    
    const progressText = screen.getByText('0%');
    expect(progressText).toBeInTheDocument();
  });

  it('handles decimal percent progress completion', () => {
    render(<ProfileProgressDisplay progress={14.14} />);
    
    const progressText = screen.getByText('14%');
    expect(progressText).toBeInTheDocument();
  });

  it('renders with the correct size', () => {
    render(<ProfileProgressDisplay progress={50} size="8rem" />);
    
    // Check if the outer box has the correct size
    const outerBox = screen.getByTestId('outer-box');
    expect(outerBox).toHaveStyle('borderRadius: 8rem');
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<ProfileProgressDisplay progress={75} />);
    expect(asFragment).toMatchSnapshot();
  });
});