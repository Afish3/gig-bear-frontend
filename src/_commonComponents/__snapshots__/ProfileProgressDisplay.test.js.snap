// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileProgressDisplay Component matches snapshot 1`] = `[Function]`;
